### PR TITLE
Add note explaining how dispatch args and workgroup sizes interact

### PR DESCRIPTION
**Pull request by toji**
This has proven to be a point of confusion for several developers (including myself), and it does differ from the behavior of some older compute APIs, so a clarifying note seems appropriate.

***
<a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/pull/2519.html" title="Last updated on Jan 24, 2022, 4:55 PM UTC (cf2eb60)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2519/66d765b...cf2eb60.html" title="Last updated on Jan 24, 2022, 4:55 PM UTC (cf2eb60)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -7357,6 +7357,19 @@ dictionary GPUComputePassDescriptor : GPUObjectDescriptorBase {
         </div>
 </dl>
 
+<div class=note>
+Note: The `x`, `y`, and `z` values passed to {{GPUComputePassEncoder/dispatch()}} and
+{{GPUComputePassEncoder/dispatchIndirect()}} are the number of _workgroups_ do dispatch for each
```
Comment (jrprice) [inline spec/index.bs:7362]: s/do/to

File: spec/index.bs [hunk 2]
```diff
@@ -7357,6 +7357,18 @@ dictionary GPUComputePassDescriptor : GPUObjectDescriptorBase {
...
+dimension, *not* the number of shader invocations to perform across each dimension. This matches the
+behavior of modern native GPU APIs, but differs from the behavior of OpenCL.
+
+This means that if a {{GPUShaderModule}} defines an entry point with `@workgroup_size(4, 4)`, and
+work is dispatched to it with the call `computePass.dispatch(8, 8);` the entry point will be invoked
+1024 times total: 4 dispatches of a workgroup size of 8 along both the X and Y axes.
```
Comment (jrprice) [inline spec/index.bs:7368]: Should be: 8 dispatches of a workgroup size of 4

Also, I had to re-read this bit a couple of times. Initially I read it as:
"(4 dispatches) of (a workgroup size of 8 along both the X and Y axes)"
which isn't right. Maybe there's a clearer way of phrasing this?

Comment (toji) [reply spec/index.bs]: Thanks for catching that. Took a crack at re-phrasing it, let me know what you think!

File: spec/index.bs [hunk 3]
```diff
@@ -7357,6 +7357,18 @@ dictionary GPUComputePassDescriptor : GPUObjectDescriptorBase {
         </div>
 </dl>
 
+<div class=note>
+Note: The `x`, `y`, and `z` values passed to {{GPUComputePassEncoder/dispatch()}} and
+{{GPUComputePassEncoder/dispatchIndirect()}} are the number of _workgroups_ to dispatch for each
```
Comment (kainino0x) [inline spec/index.bs:7362]:
```suggestion
{{GPUComputePassEncoder/dispatchIndirect()}} are the number of *workgroups* to dispatch for each
```
bikeshed doesn't treat underscores as markdown

**Review by kainino0x — APPROVED** [review]
How about also renaming the arguments of dispatch() (and the sample code in dispatchIndirect()) to be more specific? OpenGL for example uses num_groups_x/y/z, Vulkan groupCountX/Y/Z.

**Review by dneto0 — COMMENTED** [review]
LGTM.  Thanks!

**Review by kainino0x — APPROVED** [review]

**Pull request merged**